src/main/java/com/fasterxml/jackson/core/base/ParserBase.java

```diff
@@ -819,6 +824,80 @@ public BigDecimal getDecimalValue() throws IOException
         return _getBigDecimal();
     }
 
+    @Override
+    public LazyNumber getLazyNumber() throws IOException
+    {
+        LazyNumber lazyNumber = null;
+        if ((_numTypesValid & NR_BIGDECIMAL) == 0) {
+            if (_numTypesValid == NR_UNKNOWN) {
+                _parseNumericValue(NR_BIGDECIMAL);
```

Comment:
This seems wrong: if the underlying number is integer, it should not be coerced into `BigDecimal`?
Or do I misremember logic in `_parseNumericValue`
I guess I should have a look at unit test here first. :)

Reply:
the if stmt is `if ((_numTypesValid & NR_BIGDECIMAL) == 0)`

Reply:
Hmmh ok. What is needed here are unit tests of seeing which `LazyNumber` is actually returned in various cases. They should match `getNumber()`, 1-to-1 I think (contents that is, `Float`/`LazyFloat`, etc etc).

While there are various possible edge cases (if number value is first accessed with non-lazy, then lazy), but the most important one to me is the one where `nextToken()` is immediately followed by `getLazyNumber` call -- that should work in a way that decoding is deferred if not yet done, but eagerly decoded value (for `int` and `long`) is used otherwise.
And I think we can actually count on "short" (int, long) integers to be eagerly decoded in that way I think.

Reply:
Ok, from later comments it does look like `getLazyNumber()` would always give `LazyBigInteger` for `VALUE_NUMBER_INT` token. If so, test should verify that (assuming we don't want to add `LazyInteger` / `LazyLong`).

Reply:
@pjfanning Right, but I think that just means that no `BigDecimal` value has been decoded for current token -- but we do not automatically always want `BigDecimal`.

I think this should become clear with unit tests checking expected behavior: what I am worried about is that what is a short integral number (`int`) would become `LazyBigDecimal` here.